web/src/components/Dropdown/index.tsx

```diff
@@ -45,21 +46,35 @@ export interface DropdownComponentProps {
  */
 
 export const DropdownComponent = (props: DropdownComponentProps) => {
-    const [selected, setSelected] = React.useState<string | undefined>('');
+    const [selected, setSelected] = useState<string | undefined>('');
 
     const { list, className, placeholder, clear, onSelect } = props;
     const defaultPlaceholder = list[0];
 
-    const cx = React.useMemo(() => classnames('cr-dropdown', className, {
+    const cx = useMemo(() => classnames('cr-dropdown', className, {
         'cr-dropdown--default': selected === placeholder,
     }), [selected, placeholder, className]);
 
-    const handleSelect = React.useCallback((elem: DropdownElem, index: number) => {
+    useEffect(() => {
+        if (clear !== false) {
+            setSelected(placeholder || convertToString(defaultPlaceholder));
+        }
+    }, [placeholder, defaultPlaceholder, clear]);
+
+    useEffect(() => {
+        if (typeof props.selectedValue !== 'undefined') {
+            setSelected(props.selectedValue);
+        } else if (props.selectedValue === '') {
+            setSelected(placeholder || convertToString(defaultPlaceholder));
+        }
+    }, [props.selectedValue]);
```

Comment:
1. Combine these 2 useEffects into one as it affects to the same variable.
2. We have type checks in the `convertToString`. This should be ok:
```suggestion
    useEffect(() => {
        if (clear !== false) {
            setSelected(convertToString(placeholder || defaultPlaceholder)));
        }
    }, [placeholder, defaultPlaceholder, clear, props.selectedValue]);
```